### ISSUE TITLE
hive dialect: "Found unparsable section" when CREATE statement contains PRIMARY KEY(column) DISABLE NOVALIDATE

### ISSUE THREAD

**Issue by rodmax** [sqlfluff/sqlfluff#4042]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I get error when my HIVE script contains `CREATE` statement with `PRIMARY KEY(id) DISABLE NOVALIDATE`:

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE my_data_lint_error (\n id...'
WARNING: Parsing errors found and dialect is set to 'hive'. Have you configured your dialect correctly?
```

### Expected Behaviour

`sqlfluff` should parse `PRIMARY KEY(column) DISABLE NOVALIDATE` **Hive** statement

### Observed Behaviour

The Error only appears for construction `PRIMARY KEY(column) DISABLE NOVALIDATE`
💡 if remove `DISABLE NOVALIDATE` modifiers then the error disappears

```sql
-- parse ERROR
CREATE TABLE my_data_lint_error (
    id BIGINT,
    PRIMARY KEY(id) DISABLE NOVALIDATE
)

-- everything OK
CREATE TABLE my_data_lint_passed (
    id BIGINT,
    PRIMARY KEY(id)
)
```

UPDATED: it seems this syntax will be added to hive `2.1.0` version, https://issues.apache.org/jira/browse/HIVE-13290

![image](https://user-images.githubusercontent.com/6079597/200580055-718dac73-ea74-4933-8311-430a77e0c04d.png)

### How to reproduce

Create file `test.sql` with content

```sql
CREATE TABLE my_data_lint_error (
    id BIGINT,
    PRIMARY KEY(id) DISABLE NOVALIDATE
)
```

then run command
```bash
sqlfluff lint --dialect hive test.sql
```

### Dialect

`hive`

### Version

```bash
sqlfluff --version                   
sqlfluff, version 1.4.1
python --version                      
Python 3.8.11
```

### Configuration

not used

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
